### PR TITLE
HSEARCH-1812 Document how to get the SearchIntegrator

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/HSEARCH-1812

### CODE REVIEW

File: documentation/src/main/asciidoc/advanced-features.asciidoc [hunk 1]
```diff
@@ -19,6 +20,22 @@ SearchFactory searchFactory = fullTextSession.getSearchFactory();
 ----
 ====
 
+[[AccessingSearchIntegrator]]
+=== Accessing the SearchIntegrator
```
Comment (hferentschik) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: `SearchIntegrator` - back quotes for formatting missing

File: documentation/src/main/asciidoc/advanced-features.asciidoc [hunk 2]
```diff
@@ -19,6 +20,22 @@ SearchFactory searchFactory = fullTextSession.getSearchFactory();
...
+The SearchIntegrator is an object that gives access to lower level API of Hibernate Search.
```
Comment (hferentschik) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: The interface `SearchIntegrator` gives access to lower level APIs of Hibernate Search.

File: documentation/src/main/asciidoc/advanced-features.asciidoc [hunk 3]
```diff
@@ -19,6 +20,22 @@ SearchFactory searchFactory = fullTextSession.getSearchFactory();
...
+Frameworks can use it to integrate with Hibernate Search. It should not be exposed as a
+public API.
```
Comment (hferentschik) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: "It should not be exposed as a public API"  - not sure what this is supposed to mean. It sounds like a recommendation to the user to now expose it as public API, but that is really not in the hands of the users. Given its package name (`org.hibernate.search.spi`), I would just say it is a SPI.

File: documentation/src/main/asciidoc/advanced-features.asciidoc (hunk 1 above)
Comment (DavideD) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: `IndexReader` and `SearchFactory` appear in the same page and none of them use back quotes for formatting. Should it be changed everywhere?

File: documentation/src/main/asciidoc/advanced-features.asciidoc (hunk 3 above)
Comment (DavideD) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: I don't understand how

> It should **not** be exposed as a public API

can be confused as a recommendation  to the user to now expose it as public API.

I will just remove this sentence anyway, it's related to frameworks and I don't think it's needed here.

File: documentation/src/main/asciidoc/advanced-features.asciidoc (hunk 1 above)
Comment (Sanne) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: Yes ideally it should. But let's focus on your fix now.. if you want to change the others too, that's great but keep it as as a separate commit.

**Comment by DavideD** [comment]
I've updated the documentation and added the backquotes where missing in a separate commit

**Comment by DavideD** [comment]
Reopend...I've pressed the wrong button

File: documentation/src/main/asciidoc/advanced-features.asciidoc (hunk 3 above)
Comment (hferentschik) [inline documentation/src/main/asciidoc/advanced-features.asciidoc]: > I will just remove this sentence anyway, it's related to frameworks and I don't think it's needed here.

+1

**Comment by hferentschik** [comment]
Re-based and merged
